packages/Config/RectorConfig.php

```diff
@@ -163,14 +168,28 @@ public function ruleWithConfiguration(string $rectorClass, array $configuration)
         Assert::isAOf($rectorClass, RectorInterface::class);
         Assert::isAOf($rectorClass, ConfigurableRectorInterface::class);
 
-        $this->singleton($rectorClass);
-        $this->afterResolving($rectorClass, static function (ConfigurableRectorInterface $configurableRector) use (
+        // store configuration to cache
+        $this->ruleConfigurations[$rectorClass] = array_merge(
+            $this->ruleConfigurations[$rectorClass] ?? [],
             $configuration
-        ): void {
-            $configurableRector->configure($configuration);
-        });
+        );
+
+        $isBound = $this->bound($rectorClass);
+
+        // avoid double registration
+        if ($isBound) {
+            return;
+        }
 
+        $this->singleton($rectorClass);
         $this->tagRectorService($rectorClass);
```

Comment:
Without this check, the same Rector rule would be registered twice. That could lead to unexpected changes and waste of resources.